### PR TITLE
Implement UI for adding/editing/deleting cheats

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
Screenshot:
image

More description can be found in the commit messages.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4610)

### CODE REVIEW

File: src/citra_qt/cheats.cpp [hunk 1]
```diff
@@ -56,20 +67,186 @@ void CheatDialog::LoadCheats() {
...
+        this, tr("Cheats"), tr("Would you like to save the current cheat?"),
+        QMessageBox::Yes | QMessageBox::No | QMessageBox::Cancel, QMessageBox::Cancel);
+
+    if (answer == QMessageBox::Yes) {
+        return SaveCheat(last_row);
+    } else if (answer == QMessageBox::Cancel) {
+        return false;
+    } else {
+        return true;
+    }
```
Comment (FearlessTobi) [inline src/citra_qt/cheats.cpp]: Can't this be:
```
    if (answer == QMessageBox::Yes) {
        return SaveCheat(last_row);
    } else {
        return answer != QMessageBox::Cancel;
    }
```
?

File: src/citra_qt/cheats.h [hunk 2]
```diff
@@ -19,12 +23,38 @@ class CheatDialog : public QDialog {
...
 private:
-    std::unique_ptr<Ui::CheatDialog> ui;
-
+    /**
+     * Loads the cheats from the CheatEngine, and populates the table.
+     */
     void LoadCheats();
 
+    /**
+     * Pops up a message box asking if the user want to save the current cheat.
```
Comment (FearlessTobi) [inline src/citra_qt/cheats.h]: "Pops up a message box asking if the user want**s** to save the current cheat."

File: src/citra_qt/cheats.h [hunk 3]
```diff
@@ -19,12 +23,38 @@ class CheatDialog : public QDialog {
...
+     * If the user selected Yes, attempts to save the current cheat.
+     * @return true if the user selected No, or if the cheat is saved successfully
```
Comment (FearlessTobi) [inline src/citra_qt/cheats.h]: "if the cheat **was** saved successfully"?

**Comment by zhaowenlan1779** [comment]
will merge soon if no more comments.

**Pull request merged**